tests/iio_attr.c

```diff
@@ -528,21 +663,33 @@ int main(int argc, char **argv)
 		return EXIT_FAILURE;
 	if (wbuf && !wbuf)
 		return EXIT_FAILURE;
-	if (wbuf && ((device_index && (!strcmp(".", argv[device_index]) ||
+	if ((gen_code || wbuf) && ((device_index && (!strcmp(".", argv[device_index]) ||
 				        strchr(argv[device_index], '*'))) ||
 		     (channel_index && (!strcmp(".", argv[channel_index]) ||
 					 strchr(argv[channel_index], '*'))) ||
 		     (attr_index && (!strcmp(".", argv[attr_index])  ||
 				      strchr(argv[attr_index], '*'))))) {
-		printf("can't write value with wildcard match\n");
+		printf("can't %s with wildcard match\n",
+				gen_code ? "generate code" : "write value");
 		return EXIT_FAILURE;
 	}
 
+
+	if (gen_code) {
+		gen_fd = fopen (gen_file, "w");
+		if (!gen_fd) {
+			errnum = errno;
```

Comment:
`errnum` is only used in this block
it does not seem to be needed, since it only gets `errno` assigned and then this block exits;
it could be removed;